src/client/componentsMealsharing/PostForm.js

```diff
@@ -0,0 +1,85 @@
+import React,{Component,useState} from "react";
+
+export function PostForm() {
+    const [seat, setSeat] = useState("");
+    const [phone, setPhone] = useState("");
+    const [name, setName] = useState("");
+    const [mail, setMail] = useState("");
+    const [reservation, setReservation] = useState([]);
+    bookSeats = ()=> {
+       setReservation (prev=>{[
+         ...prev,
+         {
+           number_of_guests: seat,
+           // created_date: request.body.created_date,
+           contact_phonenumber: phone,
+           contact_name: name,
+           contact_email: mail,
+           // meal_id: request.body.meal_id,
+           // number_of_guests: seat,
+         },
+       ];})
+
+       const options = {
+         method: "POST",
+         headers: {
+           "Content-Type": "application/json",
+         },
+         body: JSON.stringify(data)
+       };
+       fetch("http://localhost:3000/api/reservations", options);
+    }
+    return (
+      <form onSubmit={bookSeats}>
+        <div>
+          <label for="seat">seats : </label>
+          <input
+            type="num"
+            id="seat"
+            name="seat"
+            placeholder="No of guests"
+            value={seat}
```

Comment:
and good points if you add " required " to input elements 
that will make sure the user have to fill it out